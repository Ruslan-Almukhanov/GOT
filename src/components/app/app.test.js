import React from 'react';
import App from './app';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Testing <App/>', () => {
    it('check <App/> rendering', async () => {
        const app = await render(<App/>)
        expect(app).toMatchSnapshot();
    })
    it('check toggle button in the document', async () => {
        const {getByText} = await render(<App/>)
        const btn = getByText(/Random character/)
        expect(btn).toBeTruthy()
    })
})