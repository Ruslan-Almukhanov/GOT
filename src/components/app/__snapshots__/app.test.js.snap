// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <App/> check <App/> rendering 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="container"
        >
          <div
            class="sc-bdVaJa euyOib"
          >
            <h3
              class="sc-bwzfXH bmIlQj"
            >
              <a
                href="/"
              >
                Game of Thrones DB
              </a>
            </h3>
            <ul
              class="sc-htpNat jvnEHg"
              data-testid="test-menu"
            >
              <li>
                <a
                  href="/characters/"
                >
                  Characters
                </a>
              </li>
              <li>
                <a
                  href="/houses/"
                >
                  Houses
                </a>
              </li>
              <li>
                <a
                  href="/books/"
                >
                  Books
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-lg-5 offset-lg-0"
            >
              <div
                class="random-block rounded"
              >
                <div
                  class="loadingio-spinner-eclipse-4yryhxszik"
                >
                  <div
                    class="ldio-yfe4bbbunhb"
                  >
                    <div />
                  </div>
                </div>
              </div>
              <button
                class="btn-toggle"
              >
                Random character
              </button>
            </div>
          </div>
          <h1
            style="color: rgb(255, 255, 255); text-align: center;"
          >
            Welcome to the Game Of Thrones !!!
          </h1>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="container"
      >
        <div
          class="sc-bdVaJa euyOib"
        >
          <h3
            class="sc-bwzfXH bmIlQj"
          >
            <a
              href="/"
            >
              Game of Thrones DB
            </a>
          </h3>
          <ul
            class="sc-htpNat jvnEHg"
            data-testid="test-menu"
          >
            <li>
              <a
                href="/characters/"
              >
                Characters
              </a>
            </li>
            <li>
              <a
                href="/houses/"
              >
                Houses
              </a>
            </li>
            <li>
              <a
                href="/books/"
              >
                Books
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-lg-5 offset-lg-0"
          >
            <div
              class="random-block rounded"
            >
              <div
                class="loadingio-spinner-eclipse-4yryhxszik"
              >
                <div
                  class="ldio-yfe4bbbunhb"
                >
                  <div />
                </div>
              </div>
            </div>
            <button
              class="btn-toggle"
            >
              Random character
            </button>
          </div>
        </div>
        <h1
          style="color: rgb(255, 255, 255); text-align: center;"
        >
          Welcome to the Game Of Thrones !!!
        </h1>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
